python/libcuvs/pyproject.toml

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+[build-system]
+
+requires = [
+    "rapids-build-backend>=0.3.0,<0.4.0.dev0",
+    "scikit-build-core[pyproject]>=0.10.0",
+] # This list was generated by `rapids-dependency-file-generator`. To make changes, edit ../../dependencies.yaml and run `rapids-dependency-file-generator`.
+build-backend = "rapids_build_backend.build"
+
+[project]
+name = "libcuvs"
+dynamic = ["version"]
+description = "cuVS: Vector Search on the GPU (C++)"
+readme = { file = "README.md", content-type = "text/markdown" }
+authors = [
+    { name = "NVIDIA Corporation" },
+]
+license = { text = "Apache 2.0" }
+requires-python = ">=3.10"
+dependencies = [
+    "libraft==25.2.*,>=0.0.0a0",
+    "nvidia-cublas",
+    "nvidia-curand",
+    "nvidia-cusolver",
+    "nvidia-cusparse",
+] # This list was generated by `rapids-dependency-file-generator`. To make changes, edit ../../dependencies.yaml and run `rapids-dependency-file-generator`.
+classifiers = [
+    "Intended Audience :: Developers",
+]
+
+[project.urls]
+Homepage = "https://github.com/rapidsai/cuvs"
+Documentation = "https://docs.rapids.ai/api/cuvs/stable/"
+
+[project.entry-points."cmake.prefix"]
+libcuvs = "libcuvs"
+
+[tool.isort]
```

Comment:
Do we need all the isort settings here?

Reply:
Narrowed these settings down in https://github.com/rapidsai/cuvs/pull/594/commits/4724d800d032104a7f669088312b7c997c91b2fc